### ISSUE TITLE
Stage1 publishing copies binaries over to Stage2

### ISSUE THREAD

**Issue by Sridhar-MS** [dotnet/cli#484]
> *** Building stage1 dotnet using downloaded stage0 ... ***
15:56:44 Publishing [33mdotnet[39m for [33mDNXCore,Version=v5.0[39m/[33mubuntu.14.04-x64[39m
15:56:46 Compiling [33mMicrosoft.DotNet.ProjectModel[39m for [33mDNXCore,Version=v5.0[39m
15:56:48 Copy From: /mnt/j/workspace/dotnet_cli/debug_ubuntu/artifacts/ubuntu.14.04-x64/stage1/bin/Microsoft.DotNet.ProjectModel.dll
15:56:48        To: /mnt/j/workspace/dotnet_cli/debug_ubuntu/artifacts/ubuntu.14.04-x64/stage2/bin/
15:56:48 Copy From: /mnt/j/workspace/dotnet_cli/debug_ubuntu/artifacts/ubuntu.14.04-x64/stage1/bin/Microsoft.DotNet.ProjectModel.pdb
15:56:48        To: /mnt/j/workspace/dotnet_cli/debug_ubuntu/artifacts/ubuntu.14.04-x64/stage2/bin/

Probably related to #444.

**Comment by piotrpMSFT** [comment]
@cdmihai is this something you want to pick up while you're thinking about eliminating post-compile scripts?

**Comment by cdmihai** [comment]
Yes, we were actually studying this exact behaviour yesterday when we were looking into taking them out.

We found out that the `place-binary` script always copies to stage2, indifferent of what you compile  So compilation of stage 1 copies output to stage 2. Compilation of stage2 copies output over itself. This got us a bit puzzled about the meaning of the script so we removed removed it entirely and CLI still builds itself, with everything passing.  :)

I'll put out a PR out next week (removing the scripts turns on incremental compilation and I still want to make sure that one works reasonably well ...)
